src/git-repository-async.js

```diff
@@ -0,0 +1,719 @@
+      .then(hunks => _.flatten(hunks)) // hunks :: Array<Array<Hunk>>
+  }
+
+  _getDiffLines (diff) {
+    return this._getDiffHunks(diff)
+      .then(hunks => Promise.all(hunks.map(h => h.lines())))
+      .then(lines => _.flatten(lines)) // lines :: Array<Array<Line>>
+  }
+
+  // Retrieving Diffs
+  // ================
+  // Public: Retrieves the number of lines added and removed to a path.
+  //
+  // This compares the working directory contents of the path to the `HEAD`
+  // version.
+  //
+  // * `path` The {String} path to check.
+  //
+  // Returns a {Promise} which resolves to an {Object} with the following keys:
+  //   * `added` The {Number} of added lines.
+  //   * `deleted` The {Number} of deleted lines.
+  getDiffStats (_path) {
+    return this.repoPromise
+      .then(repo => Promise.all([repo, repo.getHeadCommit()]))
+      .then(([repo, headCommit]) => Promise.all([repo, headCommit.getTree()]))
+      .then(([repo, tree]) => {
+        const options = new Git.DiffOptions()
+        options.pathspec = _path
+        return Git.Diff.treeToWorkdir(repo, tree, options)
+      })
+      .then(diff => this._getDiffLines(diff))
+      .then(lines => {
+        const stats = {added: 0, deleted: 0}
+        for (const line of lines) {
+          const origin = line.origin()
+          if (origin === Git.Diff.LINE.ADDITION) {
+            stats.added++
+          } else if (origin === Git.Diff.LINE.DELETION) {
+            stats.deleted++
+          }
+        }
+        return stats
+      })
+  }
+
+  _diffBlobToBuffer (blob, buffer, options) {
+    const hunks = []
+    const hunkCallback = (delta, hunk, payload) => {
+      hunks.push(hunk)
+    }
+    return Git.Diff.blobToBuffer(blob, null, buffer, null, null, options, null, null, hunkCallback, null, null).then(_ => hunks)
+  }
+
+  // Public: Retrieves the line diffs comparing the `HEAD` version of the given
+  // path and the given text.
+  //
+  // * `path` The {String} path relative to the repository.
+  // * `text` The {String} to compare against the `HEAD` contents
+  //
+  // Returns an {Array} of hunk {Object}s with the following keys:
+  //   * `oldStart` The line {Number} of the old hunk.
+  //   * `newStart` The line {Number} of the new hunk.
+  //   * `oldLines` The {Number} of lines in the old hunk.
+  //   * `newLines` The {Number} of lines in the new hunk
+  getLineDiffs (_path, text) {
+    return this.repoPromise
+      .then(repo => repo.getHeadCommit())
+      .then(commit => commit.getEntry(_path))
+      .then(entry => entry.getBlob())
+      .then(blob => {
+        const options = new Git.DiffOptions()
+        if (process.platform === 'win32') {
+          // Ignore eol of line differences on windows so that files checked in
+          // as LF don't report every line modified when the text contains CRLF
+          // endings.
+          options.flags = Git.Diff.OPTION.IGNORE_WHITESPACE_EOL
```

Comment:
Huh... Unfortunately I doubt it. Our [current implementation](https://github.com/atom/atom/blob/77a540962312ecf2a9cedd483cb4329564590458/src/git-repository.coffee#L387) is supposed to use this flag too.